creator-node/src/services/stateMachineManager/index.js

```diff
@@ -0,0 +1,111 @@
+const config = require('../../config')
+const { logger } = require('../../logging')
+const StateMonitoringQueue = require('./monitoring/StateMonitoringQueue')
+const NodeToSpIdManager = require('./nodeToSpIdManager')
+
+// Modes used in issuing a reconfig. Each successive mode is a superset of the mode prior.
+// The `key` of the reconfig states is used to identify the current reconfig mode.
+// The `value` of the reconfig states is used in the superset logic of determining which type of
+// reconfig is enabled.
+const RECONFIG_MODES = Object.freeze({
+  // Reconfiguration is entirely disabled
+  RECONFIG_DISABLED: {
+    key: 'RECONFIG_DISABLED',
+    value: 0
+  },
+  // Reconfiguration is enabled only if one secondary is unhealthy
+  ONE_SECONDARY: {
+    key: 'ONE_SECONDARY',
+    value: 1
+  },
+  // Reconfiguration is enabled for one secondary and multiple secondaries (currently two secondaries)
+  MULTIPLE_SECONDARIES: {
+    key: 'MULTIPLE_SECONDARIES',
+    value: 2
+  },
+  // Reconfiguration is enabled for one secondary, multiple secondaries, a primary, and a primary and one secondary
+  PRIMARY_AND_OR_SECONDARIES: {
+    key: 'PRIMARY_AND_OR_SECONDARIES',
+    value: 3
+  },
+  // Reconfiguration is enabled for one secondary, multiple secondaries, a primary, and a primary and one secondary,
+  // and entire replica set
+  // Note: this mode will probably be disabled.
+  ENTIRE_REPLICA_SET: {
+    key: 'ENTIRE_REPLICA_SET',
+    value: 4
+  }
+})
+
+const RECONFIG_MODE_KEYS = Object.keys(RECONFIG_MODES)
+
+/**
+ * Manages the queue for monitoring the state of Content Nodes and
+ * the queue for reconciling anomalies in the state (syncs and replica set updates).
+ */
+class StateMachineManager {
+  constructor() {
+    this.updateEnabledReconfigModesSet()
+    this.stateMonitoringQueue = new StateMonitoringQueue()
+  }
+
+  async init(audiusLibs) {
+    await this.stateMonitoringQueue.init(audiusLibs)
+    // TODO: Decide on interval to run this on
+    try {
+      NodeToSpIdManager.updateCnodeEndpointToSpIdMap(audiusLibs.ethContracts)
+
+      // Update enabledReconfigModesSet after successful `updateCnodeEndpointToSpIdMap()` call
+      this.updateEnabledReconfigModesSet()
+    } catch (e) {
+      // Disable reconfig after failed update
+      this.updateEnabledReconfigModesSet(
+        /* override */ RECONFIG_MODES.RECONFIG_DISABLED.key
+      )
+      logger.error(`updateEndpointToSpIdMap Error: ${e.message}`)
+    }
+  }
+
+  getStateMonitoringQueue() {
+    return this.stateMonitoringQueue.queue
+  }
```

Comment:
exposed here for service registry, which adds the queue to bull-board (the `/health/bull` endpoint)